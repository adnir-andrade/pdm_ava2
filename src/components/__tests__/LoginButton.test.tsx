import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import LoginButton from "../LoginButton";

describe("<LoginButton />", () => {
  it("renders correctly", () => {
    const { toJSON } = render(<LoginButton />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("passes father's onPress function correctly", () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(<LoginButton onPress={onPressMock} />);
    const button = getByTestId("login-button");

    fireEvent.press(button);

    expect(onPressMock).toHaveBeenCalled();
  });
});
