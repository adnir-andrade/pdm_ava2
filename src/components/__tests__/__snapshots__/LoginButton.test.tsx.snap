// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "marginTop": 20,
      "opacity": 1,
    }
  }
  testID="login-button"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "height": 60,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": 60,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
